### PR TITLE
Create Web Annotation-based Highlights from Context Menus

### PR DESCRIPTION
**Pull request by BigBlueHat**
Very early. Not usable. But very promising.

All the goodness happens in the console output for the page or the background script's console (look up how to do that for your favorite browser).

### CODE REVIEW

**Comment by BigBlueHat** [comment]
@tilgovi @treora fancy reviewing some code / this Web Extension? 😸 

Lots yet to do, but I believe this has now reached "compelling demo" status. :grin:

Cheers!
🎩

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,8 @@
 node_modules
 
 # build these don't commit them
+extension/annotate.js
+extension/background.js
+extension/highlight.js
```
Comment (Treora) [inline .gitignore:6]: `highlighter.js`

**Comment by Treora** [comment]
`git add src/highlighter.js`? :)

File: src/storage.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+// returns the PouchDB promise
+function storeAnnotation(annotation) {
+  // construcut unique collation friendly id (for _id & id)
+  // TODO: find a better URN to keep this stuff in
+  // assume annotation.target is an IRI if .source is missing
+  let target_iri = annotation.target.source || annotation.target;
+  let id = 'urn:page-notes:'
+    + encodeURI(target_iri)
+    + ':' + (new Date).toISOString();
```
Comment (Treora) [inline src/storage.js:74]: Would this blow up one day when someone makes annotations really fast after each other? Humans might not, but scripts may.

**Comment by Treora** [comment]
Nice stuff. Will try it out tomorrow.

**Pull request merged**